packages/app/src/cli/services/function/binaries.ts

```diff
@@ -10,12 +10,22 @@ import * as gzip from 'node:zlib'
 import {fileURLToPath} from 'node:url'
 
 const FUNCTION_RUNNER_VERSION = 'v6.3.0'
-const JAVY_VERSION = 'v3.2.0'
+const JAVY_VERSION = 'v4.0.0'
+const JAVY_PLUGIN_VERSION = 'v4.0.0'
+
+interface DownloadableBinary {
+  path: string
+  name: string
+  version: string
+
+  downloadUrl(processPlatform: string, processArch: string): string
+  processResponse(responseStream: PipelineSource<unknown>, outputStream: fs.WriteStream): Promise<void>
+}
 
 // The logic for determining the download URL and what to do with the response stream is _coincidentally_ the same for
```

Comment:
I think this comment is out of date -- they _have_ diverged?

Reply:
So the logic for determining the download URL and processing the response stream is the still the same for the Javy CLI and function-runner binaries. It's different for the Javy plugin, but this comment doesn't refer to the Javy plugin. 

This comment is on top of and only applies to the `Executable` class which the Javy CLI and function-runner use. The Javy plugin is downloaded with the `JavyPlugin` class, not the `Executable` class. The `Executable` and `JavyPlugin` classes both implement the `DownloadableBinary` interface but the comment is independent of that interface.